xio-core/src/test/java/com/xjeffrose/xio/config/RouteReloaderUnitTest.java

```diff
@@ -0,0 +1,293 @@
+  }
+
+  @Test(expected = IllegalStateException.class)
+  public void testInitBadValue() throws Exception {
+
+    ScheduledExecutorService executor = new ScheduledThreadPoolExecutor(1);
+    RouteReloader<TrivialFactory> subject =
+        new RouteReloader<TrivialFactory>(executor, TrivialFactory::new);
+    String initialFileContents = badValue;
+    String input = rawCreateConf(initialFileContents, inputJsonFilename);
+    TrivialFactory initialValue = subject.init(input);
+  }
+
+  @Test
+  public void testReload_WhenWatchedFilesDoNotChange() throws Exception {
+
+    AtomicBoolean fireUpdatedCalled = new AtomicBoolean(false);
+    ScheduledExecutorService executor = new ScheduledThreadPoolExecutor(1);
+
+    RouteReloader<TrivialFactory> subject =
+        new RouteReloader<TrivialFactory>(executor, TrivialFactory::new);
+    String initialFileContents = helloWorldValue;
+    String input = rawCreateConf(initialFileContents, inputJsonFilename);
+    // init the subject with an file that contains the helloworldvalue
+    TrivialFactory initialValue = subject.init(input);
+    // the result from the init should be the baseline TrivialFactory output and it should match the helloworldvalue
+    assertEquals(initialFileContents, initialValue.inputString);
+
+    TrivialState state =
+        new TrivialState() {
+          @Override
+          public void fireUpdated() {
+            fireUpdatedCalled.set(true);
+          }
+        };
+
+    // start the threaded update
+    subject.start(state::update);
+    Thread.sleep(5000);
+    // since we did not change any files we should not get an update
+    assertFalse(fireUpdatedCalled.get());
+    // the update should have never been called and these values should not be set
+    assertEquals(null, state.oldValue);
+    assertEquals(null, state.newValue);
+    executor.shutdown();
+  }
+
+  @Test
+  public void testReload_WhenWatchedFilesChange_Date_Was_Modified_and_Digest_Was_NOT_Changed()
+      throws Exception {
+
+    AtomicBoolean fireUpdatedCalled = new AtomicBoolean(false);
+    ScheduledExecutorService executor = new ScheduledThreadPoolExecutor(1);
+
+    RouteReloader<TrivialFactory> subject =
+        new RouteReloader<TrivialFactory>(executor, TrivialFactory::new);
+    String initialFileContents = helloWorldValue;
+    String input = rawCreateConf(initialFileContents, inputJsonFilename);
+    // init the subject with an file that contains the helloworldvalue
+    TrivialFactory initialValue = subject.init(input);
+    // the result from the init should be the baseline TrivialFactory output and it should match the helloworldvalue
+    assertEquals(initialFileContents, initialValue.inputString);
+
+    TrivialState state =
+        new TrivialState() {
+          @Override
+          public void fireUpdated() {
+            fireUpdatedCalled.set(true);
+          }
+        };
+
+    subject.start(state::update);
+    Thread.sleep(5000);
```

Comment:
Thread sleeps are sad and slow. Consider an `@Test(timeout=...` and and observe when the subject is ready. You could possibly use a CountDownLatch as well.

Reply:
i'll try test timeout

Reply:
I'll change this in  a later pull request